airflow-core/src/airflow/jobs/scheduler_job_runner.py

```diff
@@ -867,7 +921,14 @@ def process_executor_events(
 
                 # Get task from the Serialized DAG
                 try:
-                    dag = dag_bag.get_dag(ti.dag_id)
+                    dag = scheduler_dag_bag.get_dag(dag_run=ti.dag_run, session=session)
+                    cls.logger().error(
+                        "DAG '%s' for task instance %s not found in serialized_dag table",
+                        ti.dag_id,
+                        ti,
+                    )
```

Comment:
Since we no longer delete serdag, I think any task instance from workers will have the serdag? The exception handling is not needed anymore. Next PR though. It will help to identify issues.

Reply:
i think in order to remove this kind of exception handling, we might need to make the entities not nullable, and perhaps have cascade deletes etc because, right now, most of the relationships are optional so e.g. mypy complains that they might be None ---- and it's possible if it's not guaranteed by the database.

Reply:
And since we want this in Rc3, better to be safe rather than blow up the scheduler with an exception!